advanced/Scripts/list.sh

```diff
@@ -88,175 +88,144 @@ HandleOther() {
-        # Is it in the list? Logic follows that if its whitelisted it should not be blacklisted and vice versa
-        grep -Ex -q "${domain}" "${list}" > /dev/null 2>&1 || bool=false
-        if [[ "${bool}" == true ]]; then
-            # Remove it from the other one
-            echo -e "  ${INFO} Removing $1 from ${listname}..."
-            # /I flag: search case-insensitive
-            sed -i "/${domain}/Id" "${list}"
-            reload=true
-        else
-            if [[ "${verbose}" == true ]]; then
-                echo -e "  ${INFO} ${1} does not exist in ${listname}, no need to remove!"
-            fi
+
+    # Is the domain in the list we want to remove it from?
+    num="$(sqlite3 "${gravityDBfile}" "SELECT COUNT(*) FROM ${list} WHERE ${sqlitekey} = \"${domain}\";")"
+
+    if [[ "${num}" -ne 0 ]]; then
+        # Domain found in the file, remove it!
+        if [[ "${verbose}" == true ]]; then
+            echo -e "  ${INFO} Removing ${1} from ${listname}..."
         fi
-    elif [[ "${list}" == "${regexlist}" ]]; then
-        [[ -z "${type}" ]] && type="--wildcard-only"
-        domain="${1}"
-
-        [[ "${wildcard}" == true ]] && domain="(^|\\.)${domain//\./\\.}$"
-
-        bool=true
-        # Is it in the list?
-        grep -Fx "${domain}" "${regexlist}" > /dev/null 2>&1 || bool=false
-        if [[ "${bool}" == true ]]; then
-            # Remove it from the other one
-            echo -e "  ${INFO} Removing $domain from regex list..."
-            local lineNumber
-            lineNumber=$(grep -Fnx "$domain" "${list}" | cut -f1 -d:)
-            sed -i "${lineNumber}d" "${list}"
-            reload=true
-        else
-            if [[ "${verbose}" == true ]]; then
-                echo -e "  ${INFO} ${domain} does not exist in regex list, no need to remove!"
-            fi
+        reload=true
+        # Remove it from the current list
+        local timestamp
+        timestamp="$(date --utc +'%s')"
+        sqlite3 "${gravityDBfile}" "DELETE FROM ${list} WHERE ${sqlitekey} = \"${domain}\";"
+    else
+        if [[ "${verbose}" == true ]]; then
+            echo -e "  ${INFO} ${1} does not exist in ${list}, no need to remove!"
         fi
     fi
 }
 
-# Update Gravity
-Reload() {
-    echo ""
-    pihole -g --skip-download "${type:-}"
-}
-
 Displaylist() {
-    if [[ -f ${listMain} ]]; then
-        if [[ "${listMain}" == "${whitelist}" ]]; then
-            string="gravity resistant domains"
-        else
-            string="domains caught in the sinkhole"
-        fi
-        verbose=false
-        echo -e "Displaying $string:\n"
+    local list listname count num_pipes domain enabled status
+
+    if [[ "${listType}" == "regex" ]]; then
+        listname="regex filters list"
+    else
+        # Whitelist / Blacklist
+        listname="${listType}"
+    fi
+    data="$(sqlite3 "${gravityDBfile}" "SELECT * FROM ${listType};" 2> /dev/null)"
```

Comment:
If we're only using the domain/filter and enabled columns, then the select statement can specify those columns instead of including the date added column.

Reply:
I will also display the date added column. I intentionally left off a precise description to avoid having to differentiate between the name for the lists `domain` and for the regex table `filter`. 

@Mcat12 For the sake of simplicity, I'm more and more inclined to just call the first column of `regex` also `domain` and interested in your opinion.

Reply:
That's fine by me. The filters describe a set of domains, so it's not too much of a stretch to use `domain` as the column name.